src/app/state/tetris/tetris.state.ts

```diff
@@ -0,0 +1,130 @@
+import { PieceFactory } from '@angular-tetris/factory/piece-factory';
+import { GameState } from '@angular-tetris/interface/game-state';
+import { Piece } from '@angular-tetris/interface/piece/piece';
+import { Speed } from '@angular-tetris/interface/speed';
+import { Tile } from '@angular-tetris/interface/tile/tile';
+import { MatrixUtil } from '@angular-tetris/interface/utils/matrix';
+import { LocalStorageService } from '@angular-tetris/services/local-storage.service';
+import { DestroyRef, Injectable, computed, effect, inject, signal } from '@angular/core';
+
+export interface TetrisState {
+  matrix: Tile[];
+  current: Piece | null;
+  next: Piece;
+  hold: Piece;
+  canHold: boolean;
+  points: number;
+  locked: boolean;
+  sound: boolean;
+  initSpeed: Speed;
+  speed: Speed;
+  initLine: number;
+  clearedLines: number;
+  gameState: GameState;
+  saved: TetrisState;
+  max: number;
+}
+
+const createInitialState = (pieceFactory: PieceFactory): TetrisState => ({
+  matrix: MatrixUtil.getStartBoard(),
+  current: null,
+  next: pieceFactory.getRandomPiece(),
+  hold: pieceFactory.getNonePiece(),
+  canHold: true,
+  points: 0,
+  locked: true,
+  sound: true,
+  initLine: 0,
+  clearedLines: 0,
+  initSpeed: 1,
+  speed: 1,
+  gameState: GameState.Loading,
+  saved: null,
+  max: LocalStorageService.maxPoint
+});
+
+const isObjectShallowEqual = (a: any, b: any) => a == b;
+
+@Injectable({ providedIn: 'root' })
+export class TetrisStateService {
+  private pieceFactory = inject(PieceFactory);
+  private tetrisState = signal<TetrisState>(createInitialState(this.pieceFactory));
+
+  next = computed(() => this.tetrisState().next, {
+    equal: isObjectShallowEqual
+  });
+  hold = computed(() => this.tetrisState().hold, {
+    equal: isObjectShallowEqual
+  });
+  matrix = computed(() => this.tetrisState().matrix, {
+    equal: isObjectShallowEqual
+  });
+  current = computed(() => this.tetrisState().current, {
+    equal: isObjectShallowEqual
+  });
```

Comment:
To ensure that the additional options for `computed` work properly, it is important to define the return type. Otherwise, the type will be lost and treated as `any`.

![image](https://github.com/trungk18/angular-tetris/assets/6767322/350fd705-607b-4c53-8741-ec83cafd7aa3)

Before: `Signal<any>` 
![Screenshot 2023-07-08 at 2 43 24 PM](https://github.com/trungk18/angular-tetris/assets/6767322/fa65922b-431c-4a6a-baf1-32e1905e5a0b)

After: `Signal<Piece>`

![Screenshot 2023-07-08 at 2 46 55 PM](https://github.com/trungk18/angular-tetris/assets/6767322/17a5ad2b-ef8c-4b3d-9435-c970c96b2382)